File: gestor-fe-frontend/src/app/models/gestion.ts
import { Factura } from './factura';

export interface Gestion {
  id?: number;
  factura?: Factura;
  facturaId?: number;
  faseId: number;
  accion: 'APROBADO' | 'RECHAZADO' | string;
  estadoResultado: string; // Ej: 'EN GESTIÓN', 'CAUSADO', 'IMPUESTOS VERIFICADOS', 'PAGADO', 'ANULADO'
  tipoRegistroContableId?: number | null; // ID mapeado a la tabla maestra de tipos de registro
  numeroCausacion?: string | null;
  causalDevolucionId?: number | null;
  observacion?: string | null;
  usuario?: string | null;
  createdAt?: Date | string;
  deletedAt?: Date | string | null;
}

File: gestor-fe-backend/gestor-fe-core-service/src/main/java/com/gestor_fe/core/entity/Gestion.java
package com.gestor_fe.core.entity;

import java.time.LocalDateTime;
import jakarta.persistence.*;
import lombok.Data;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Data
@Entity
@Table(name = "gestion", schema = "gestor")
public class Gestion {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "factura_id", nullable = false)
    @JsonIgnoreProperties({"documentos", "gestiones"}) 
    private Factura factura;

    @Column(name = "fase_id", nullable = false)
    private Long faseId;

    @Column(name = "accion", nullable = false, length = 20)
    private String accion; // APROBADO / RECHAZADO

    @Column(name = "estado_resultado", nullable = false, length = 50)
    private String estadoResultado; // EN GESTIÓN, CAUSADO, IMPUESTOS VERIFICADOS, PAGADO, ANULADO

    @Column(name = "tipo_registro_contable_id", length = 10)
    private Long tipoRegistroContableId;

    @Column(name = "numero_causacion", length = 50)
    private String numeroCausacion;

    @Column(name = "causal_devolucion_id")
    private Long causalDevolucionId;

    @Column(name = "observacion", columnDefinition = "TEXT")
    private String observacion;

    @Column(name = "usuario", length = 100)
    private String usuario;

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    public void prePersist() {
        this.createdAt = LocalDateTime.now();
    }
    
    @Column(name = "deleted_at")
    private LocalDateTime deletedAt;
}